src/btchip_helpers.c

```diff
@@ -291,7 +291,7 @@ unsigned short btchip_decode_base58_address(unsigned char *in,
     return outlen;
 }
 
-void btchip_private_derive_keypair(unsigned char *bip32Path,
+int btchip_private_derive_keypair(unsigned char *bip32Path,
```

Comment:
IMO this function should be removed and replaced by:

- a function to retrieve only the pubkey based on `bip32_derive_get_pubkey_256()` from SDK crypto_helpers
- a call to `bip32_derive_ecdsa_sign_hash_256()` in btchip_sign_finalhash() instead of passing the private key here and there.